engine/src/ui/canvas/plugin.cpp

```diff
@@ -274,7 +285,7 @@ void cubos::engine::uiCanvasPlugin(Cubos& cubos)
                  Query<Entity, UICanvas&, RenderTarget&> canvasQuery, const core::io::Window& window,
                  const State& state) {
             // For each render target with a UI canvas.
-            window->renderDevice().setDepthStencilState(state.dss);
```

Comment:
You still need to set the depth stencil state! Otherwise it will use the last pass' state. Set it one where depth testing is disabled.